test/infrastructure/docker/internal/docker/machine.go

```diff
@@ -419,7 +423,65 @@ func (m *Machine) SetNodeProviderID(ctx context.Context, c client.Client) error
 	return nil
 }
 
-func (m *Machine) getKubectlNode(ctx context.Context) (*types.Node, error) {
+// CloudProviderNodePatch sets the docker provider ID for the kubernetes node.
+func (m *Machine) CloudProviderNodePatch(ctx context.Context, c client.Client, dockerMachine *infrav1.DockerMachine) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	kubectlNode, err := m.getNode(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "unable to set NodeProviderID. error getting a kubectl node")
```

Comment:
wondering if "unable to set NodeProviderID."  is misleading as it only captures a subset of what we don't do in this case